src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -98,6 +104,14 @@ protected Object readResolve() throws IOException {
         return this;
     }
 
+    /** Allows user to disable saving step info, if this imposes too high a performance overhead
+     *   by setting System Property org.jenkinsci.plugins.workflow.cps.DSL.keepStepInfo to 'false'.
+     *  This may be modified at runtime by the user if needed for debugging. */
+    private static boolean isKeepStepInfo() {
+        String prop = System.getProperty(DSL.class.getName()+".keepStepInfo");
```

Comment:
@stephenc Normally I'd agree, but there's a strong reason behind this, roughly captured in the comment.  This is explicitly a runtime property because I anticipated the risk of serialization or serialization-linked performance issues and wanted a workaround to solve them without a restart needed -- instafix for things like: https://issues.jenkins-ci.org/browse/JENKINS-43934